src/PrestaShopBundle/Form/Admin/Sell/Product/ListedCombinationType.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Sell\Product;
+
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ButtonType;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\Extension\Core\Type\MoneyType;
+use Symfony\Component\Form\Extension\Core\Type\NumberType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+
+class ListedCombinationType extends AbstractType
```

Comment:
I think this "editable rows" are actually going to be a very complex thing to deal with 😞

Reply:
any suggestion?

Reply:
I like the idea of using a FormType and a collection, this makes it easier to handle it via prototype, even if it is all handled via a JS component:
- it's easier to add/remove a field via the FormType
- no need to manually create each element for js purpose only
- we can use the twig form theme to render inputs which means they will also evolve with the updates in the theme
- it may not be available right now, but I think it will make it easier for modules to add their custom field in the combination list, they just need to change (dynamically) or override this form type

Reply:
```suggestion
class CombinationType extends AbstractType
```

Reply:
No because it's only the small form type for the list Not the full form, so we will have a much bigger CombinationType later we can rename into `CombinationRowType`

And it probably deserves a sub domain for Combination

Reply:
I prefere use of `CombinationItemType` in this case.